10up/plugin-check
# Add a 'View in code editor' link to the PHPCS warnings/errors

Closes #251 

### Description of the Change
As outlined in the issue linked below, this PR conditionally displays a _View in code editor_ link beneath each PHPCS warning/error so users can make changes to their code in the code editor in the WordPress dashboard.

<img width="1273" alt="image" src="https://github.com/10up/plugin-check/assets/5321364/2d7f334a-16b3-4d5e-8e33-45b097fc71bc">

### Open files in external IDE
Editing files can be done in an external editor using the supplied `plugin_check_validation_error_source_file_editor_url_template` filter.

Examples:

__PhpStorm__
```php
add_filter( 'plugin_check_validation_error_source_file_editor_url_template', function () {
    return 'phpstorm://open?file={{file}}&line={{line}}';
} );
```

__VS Code__
```php
add_filter( 'plugin_check_validation_error_source_file_editor_url_template', function () {
    return 'vscode://file/{{file}}:{{line}}';
} );
```

### How to test the Change
Run the Plugin Check (from this branch) on any plugin with a PHPCS warning or error and click on the 'View in code editor' to be brought over to the plugin file editor. Once there, confirm that the editor scrolls down to (and highlights) the correct line.

### Changelog Entry
Added - 'View in code editor' link beneath each PHPCS error or warning.

### Credits
Props @westonruter, @felixarntz, @mukeshpanchal27, @bordoni 

### Checklist:
